src/service-worker.ts

```diff
@@ -98,97 +86,76 @@ export async function generateServiceWorkerConfig(
   });
 
-  if (swPrecacheConfig.navigateFallback === undefined) {
+  if (workboxConfig.globDirectory === undefined) {
+    workboxConfig.globDirectory = buildRoot;
+  }
+
+  if (workboxConfig.importScripts === undefined) {
+    // Map all application routes to the entrypoint.
+    workboxConfig.importScripts =
+      ['https://storage.googleapis.com/workbox-cdn/releases/3.0.0/workbox-sw.js'];
+  }
+
+  if (workboxConfig.navigateFallback === undefined) {
     // Map all application routes to the entrypoint.
-    swPrecacheConfig.navigateFallback =
-        path.relative(project.config.root, project.config.entrypoint);
+    workboxConfig.navigateFallback =
+      path.relative(project.config.root, project.config.entrypoint);
   }
 
-  if (swPrecacheConfig.navigateFallbackWhitelist === undefined) {
+  if (workboxConfig.navigateFallbackWhitelist === undefined) {
     // Don't fall back to the entrypoint if the URL looks like a static file.
     // We want those to 404 instead, since they are probably missing assets,
     // not application routes. Note it's important that this matches the
     // behavior of prpl-server.
-    swPrecacheConfig.navigateFallbackWhitelist = [hasNoFileExtension];
+    workboxConfig.navigateFallbackWhitelist = [hasNoFileExtension];
   }
 
-  if (swPrecacheConfig.directoryIndex === undefined) {
-    // By default, sw-precache maps any path ending with "/" to "index.html".
-    // This is a reasonable default for matching application routes, but 1) our
-    // entrypoint might not be called "index.html", and 2) this case is already
-    // handled by the navigateFallback configuration above. Simplest to just
-    // disable this feature.
-    swPrecacheConfig.directoryIndex = '';
-  }
-
-  // swPrecache will determine the right urls by stripping buildRoot.
-  // NOTE:(usergenic) sw-precache generate() apparently replaces the
-  // prefix on an already posixified version of the path on win32.
-  //
-  // We include a trailing slash in `stripPrefix` so that we remove leading
-  // slashes on the pre-cache asset URLs, hence producing relative URLs
-  // instead of absolute. We want relative URLs for builds mounted at non-root
-  // paths. Note that service worker fetches are relative to its own URL.
-  swPrecacheConfig.stripPrefix = addTrailingSlash(posixifyPath(buildRoot));
-
   if (options.basePath) {
-    // TODO Drop this feature once CLI doesn't depend on it.
-    let replacePrefix = options.basePath;
-    if (!replacePrefix.endsWith('/')) {
-      replacePrefix = replacePrefix + '/';
-    }
-    if (swPrecacheConfig.replacePrefix) {
-      console.info(
-          `Replacing service worker configuration's ` +
-          `replacePrefix option (${swPrecacheConfig.replacePrefix}) ` +
-          `with the build configuration's basePath (${replacePrefix}).`);
-    }
-    swPrecacheConfig.replacePrefix = replacePrefix;
+    workboxConfig.manifestTransforms = [(entries: ManifestEntry[]) => {
```

Comment:
I think you can get away with just setting

```js
workboxConfig.modifyUrlPrefix = {
  '': options.basePath
}
```

(Translation: replace the empty string as a prefix with the `options.basePath` value.)

Reply:
Done